sdk/lib/rtl/path.c

```diff
@@ -446,20 +446,144 @@ NTAPI
 RtlpApplyLengthFunction(IN ULONG Flags,
                         IN ULONG Type,
                         IN PVOID UnicodeStringOrUnicodeStringBuffer,
-                        IN PVOID LengthFunction)
+                        IN NTSTATUS(NTAPI* LengthFunction)(ULONG, PUNICODE_STRING, PULONG))
 {
-    UNIMPLEMENTED;
-    return STATUS_NOT_IMPLEMENTED;
+    NTSTATUS Status;
+    PUNICODE_STRING String;
+    ULONG Length;
+
+    if (Flags)
+    {
+        DPRINT1("ERROR: unhandled Flags = 0x%x\n", Flags);
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    if (UnicodeStringOrUnicodeStringBuffer == NULL)
+    {
+        DPRINT1("ERROR: UnicodeStringOrUnicodeStringBuffer == NULL\n");
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    if (Type == sizeof(UNICODE_STRING))
+    {
+        String = (PUNICODE_STRING)UnicodeStringOrUnicodeStringBuffer;
+    }
+    else if (Type == sizeof(RTL_UNICODE_STRING_BUFFER))
+    {
+        String = &((PRTL_UNICODE_STRING_BUFFER)UnicodeStringOrUnicodeStringBuffer)->String;
+    }
+    else
+    {
+        DPRINT1("ERROR: Type = %u\n", Type);
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    Length = 0;
+    Status = LengthFunction(0, String, &Length);
```

Comment:
The Rtlp function doesn't check for LengthFunction being the NULL pointer?

Reply:
it does, guess I crashed it on something else.